Lib/test/test_pydoc/test_pydoc.py

```diff
@@ -742,6 +742,49 @@ def run_pydoc_for_request(request, expected_text_part):
         run_pydoc_for_request(pydoc.Helper.help, 'Help on function help in module pydoc:')
         # test for pydoc.Helper() instance skipped because it is always meant to be interactive
 
+    @unittest.skipIf(hasattr(sys, 'gettrace') and sys.gettrace(),
+                     'trace function introduces __locals__ unexpectedly')
+    @requires_docstrings
+    def test_help_output_pager(self):
+        self.maxDiff = None
+
+        def run_pydoc_pager(request, what, expected_first_line):
+            with captured_output('stdout') as output, \
+                 captured_output('stderr') as err, \
+                 unittest.mock.patch('pydoc.pager') as pager_mock, \
+                 self.subTest(repr(request)):
```

Comment:
nit: I'd prefer to avoid backslashes here

```suggestion
            with (
                captured_output('stdout') as output,
                captured_output('stderr') as err,
                unittest.mock.patch('pydoc.pager') as pager_mock,
                self.subTest(repr(request)
            ):
```